src/aws.js

```diff
@@ -12,7 +12,8 @@ function buildUserDataScript(githubRegistrationToken, label) {
       `cd "${config.input.runnerHomeDir}"`,
       'export RUNNER_ALLOW_RUNASROOT=1',
       'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
-      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
+      'export INSTANCE_ID=$(wget -q -O - http://169.254.169.254/latest/meta-data/instance-id)',
+      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --name "$INSTANCE_ID" --labels ${label}`,
```

Comment:
We could maybe add the `--unattended` flag here?

Reply:
I didn't want to overload this PR with 2 many different changes. We haven't tested it with --unattended, only without and only seen it hang in one case. I don't know if it has any other consequences. I think that would be better to do in a follow PR and probably as part of the ephemeral runners work.

Reply:
Move to a ticket?